docs/topics/http/middleware.txt

```diff
@@ -340,14 +340,12 @@ Here's an example of how to create a middleware function that supports both::
         if asyncio.iscoroutinefunction(get_response):
             async def middleware(request):
                 # Do something here!
-                response = await get_response(request)
-                return response
+                return await get_response(request)
 
         else:
             def middleware(request):
                 # Do something here!
-                response = get_response(request)
-                return response
+                return get_response(request)
```

Comment:
I would revert this change. As far as I'm aware there is an intention to do something with the response in between `get_response()` and `return`, so separate statements make more sense.

Reply:
Reverted.

Reply:
> I would revert this change. As far as I'm aware there is an intention to do something with the response in between `get_response()` and `return`, so separate statements make more sense.

If this is the case we can move `# Do something here!` between `get_response()` and `return`

Reply:
I thought about that, but realised that really you'd be wanting something like:
```
            def middleware(request):
                # Do something before processing the view...
                response = get_response(request)
                # Do something after processing the view...
                return response
```
And then I decided that it felt beyond the scope of what I was trying to do. If we were to do that I'd make it a separate commit. Let me know if you'd like this change, @felixxm.

Reply:
> And then I decided that it felt beyond the scope of what I was trying to do. If we were to do that I'd make it a separate commit. Let me know if you'd like this change, @felixxm.

The current wording works for me, I'd leave it as it is :shrug: